ec2/elasticloadbalancer.py

```diff
@@ -0,0 +1,303 @@
+                                  .format(str(e)))
+
+    ctx.logger.info(
+        'Load Balancer {0} deleted. '
+        .format(elb_name))
+    if 'elb_name' in ctx.instance.runtime_properties:
+        ctx.instance.runtime_properties.pop('elb_name')
+
+    return
+
+
+def _create_health_check(user_health_check):
+
+    health_check = {
+        'interval': constants.HEALTH_CHECK_INTERVAL,
+        'healthy_threshold': constants.HEALTH_CHECK_HEALTHY_THRESHOLD,
+        'timeout': constants.HEALTH_CHECK_TIMEOUT,
+        'unhealthy_threshold': constants.HEALTH_CHECK_UNHEALTHY_THRESHOLD
+    }
+
+    health_check.update(user_health_check)
+
+    health_check = HealthCheck(**health_check)
+    return health_check
+
+
+def _create_elb_params():
+    params_dict = {'listeners': ctx.node.properties['listeners'],
+                   'name': ctx.node.properties['elb_name'],
+                   'zones': ctx.node.properties['zones']}
+    if 'security_groups' in ctx.node.properties.keys():
+        params_dict['security_groups'] = ctx.node.properties['security_groups']
+    return params_dict
+
+
+def _create_elb():
+
+    ctx.logger.info('Attempting to Create Load Balancer.')
+
+    params_dict = _create_elb_params()
+
+    try:
+        elb_client = connection.ELBConnectionClient().client()
+        lb = elb_client.create_load_balancer(**params_dict)
+    except (boto.exception.EC2ResponseError,
+            boto.exception.BotoServerError,
+            boto.exception.BotoClientError) as e:
+        raise NonRecoverableError('Load Balancer not created '
+                                  '{0}'.format(str(e)))
+
+    ctx.logger.info('Load Balancer Created. ')
+
+    utils.set_external_resource_id(value=params_dict['name'],
+                                   ctx_instance=ctx.instance,
+                                   external=False)
+
+    ctx.instance.runtime_properties['elb_name'] = params_dict['name']
+
+    if not lb:
+        raise NonRecoverableError('Load Balancer not created')
+
+    return lb
+
+
+def _get_instance_list():
+
+    ctx.logger.info('Attempting to get Load Balancer Instance List.')
+
+    elb_name = ctx.node.properties['elb_name']
+    lb = _get_existing_elb(elb_name)
+
+    try:
+        list_of_instances = lb.instances
+    except (boto.exception.EC2ResponseError,
+            boto.exception.BotoServerError,
+            boto.exception.BotoClientError) as e:
+        raise NonRecoverableError('Load Balancer instance list not loaded'
+                                  '{0}'.format(str(e)))
+
+    ctx.logger.info('retrieved Load Balancer {0} Instance List.'
+                    ' '.format(elb_name))
+
+    ctx.instance.runtime_properties[constants.EXTERNAL_RESOURCE_ID] = lb.name
+
+    return list_of_instances
+
+
+def _get_elbs_by_names(list_of_names):
+
+    ctx.logger.info(
+        'Attempting to get Load Balancers {0}.'.format(list_of_names))
+
+    try:
+        elb_client = connection.ELBConnectionClient().client()
```

Comment:
but running the client is accessing boto config, which is an external resource, so it can fail, no ?